### PR TITLE
feat: removed reliance on terraform state for config hashes

### PR DESCRIPTION
**Pull request by demeyerthom**
Removed storing the component config hash in terraform state, and instead store it in a `.mach/hashes.json` file. 

Because reading state requires a terraform init for every component before actually checking for changes the actual apply process is very slow. By storing the hashes into a file on disk we can speed up the process enormously as we can now skip many of the inits up front.

### CODE REVIEW

File: internal/hash/handler.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package hash
+
+import (
+	"context"
+	"github.com/mach-composer/mach-composer-cli/internal/config"
+	"github.com/mach-composer/mach-composer-cli/internal/graph"
+	"os"
+)
+
+const defaultHashFile = ".mach/hashes.json"
```
Comment (mvantellingen) [inline internal/hash/handler.go:10]: Should we use `.mach-composer/hashes.json` ?

File: internal/hash/handler.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+type Handler interface {
+	Store(ctx context.Context, n graph.Node) error
+	Fetch(ctx context.Context, n graph.Node) (string, error)
+}
+
+func Factory(_ *config.MachConfig) Handler {
+	hashFile := os.Getenv("HASH_FILE")
```
Comment (mvantellingen) [inline internal/hash/handler.go:18]: would it make sense to prefix the ENV?, e.g. MC_HASH_FILE ?

**Review by mvantellingen — APPROVED** [review]

File: internal/hash/handler.go (hunk 1 above)
Comment (demeyerthom) [reply internal/hash/handler.go:10]: I think we used `.mach` in the removed lockfile thing, but I will change it to mach-composer instead

File: internal/hash/handler.go (hunk 2 above)
Comment (demeyerthom) [reply internal/hash/handler.go:18]: Ah makes sense yes. I guess I should look into centralizing these and the `MCC_` variables with viper to make it a bit more clear what is coming from where. Will change this here though!

File: internal/hash/handler.go (hunk 1 above)
Comment (demeyerthom) [reply internal/hash/handler.go:10]: ah it got used by the github checkout we do. So the usage of this folder is a bit spread out

**Pull request merged**